src/projects/TodoList/index.js

```diff
@@ -43,20 +46,45 @@ class TodoList extends React.Component {
     // taskListToBeAltered.splice(i, 1, alteredTask);
     this.setState({ todos: taskListToBeAltered });
   }
+  showEditField(i) {
+    let taskListToBeAltered = this.state.todos;
+    let alteredTask = taskListToBeAltered[i];
+    alteredTask.isEditing = !alteredTask.isEditing;
+    this.setState({ todos: taskListToBeAltered });
+    this.setState({ editedItem: alteredTask.task });
+  }
+  confirmEditTask(i) {
+    let taskListToBeAltered = this.state.todos;
+    let alteredTask = taskListToBeAltered[i];
+    alteredTask.task = this.state.editedItem;
+    this.setState({ todos: taskListToBeAltered });
+    alteredTask.isEditing = !alteredTask.isEditing;
```

Comment:
Why is this modification for alteredTask placed after the setState? You're half modifying the task, setting it into state, then adding the isEditing thing after you've set it into state.

If this still works as intended it's because React is really clever under the hood about when it actually dispatches setState calls - you should move the line

Reply:
cool, will change